### ISSUE TITLE
underscore being ignored in editor.wordSeparators

### ISSUE THREAD

**Issue by MilesMcBain** [VSCodeVim/Vim#6052]
**Describe the bug**

Underscores, "_", cannot be used as word separators.

**To Reproduce**

1.  Go to settings and change `Editor: Word Separators` from default:

```
`~!@#$%^&*()-=+[{]}\|;:'",.<>/?
```

to 

```
`~!@#$%^&*()-=+[{]}\|;:'",.<>/?_
```

(underscore added at end)

2.  Open a text editor pane and add the text: `word1_word2`
3.  In edit mode, place the cursor on the first "w" of the added in step prior and issue the vim "delete word" command with keystrokes <kbd>d w</kbd>
4.  Observe the entire string is deleted.
5. In edit mode, issue the undo command, <kbd>u</kbd> to restore the string from step 2.
6. Again with the cursor on the first "w" of the string created in step 2, open the VSCode command palette and issue the command `Delete Word`  (`@command:deleteInsideWord`). 
7. Observe that only the text `word1` is deleted, leaving the remainder of the test string as `_word2`.

**Expected behavior**
Underscores should be able to be defined as word boundaries.

**Screenshots**
NA

**Environment (please complete the following information):**
- Extension Name: `vim`
- Extension Version: `1.18.5`
- OS Version: `Windows_NT x64 10.0.18363`
- VSCode version: `1.52.1`

**Additional context**
NA

**Comment by x-ji** [comment]
This doesn't seem to work using `editor.wordSeparators`, but using `vim.isKeyword` seems to work.

**Comment by x-ji** [comment]
That's true. Not sure why it didn't seem to work when I tried it yesterday.

**Comment by jdthorpe** [comment]
Seems to work when applied globally (i.e. at the root of `settings.json`) but not when used in a language specific setting (as in ` "[vim]": { "editor.wordSeparators": "`~!@#$%^&*()-=+[{]}\\|;:'\",.<>/?_" }`).  Probably related to #8190

**Comment by marcospgp** [comment]
I added `"editor.wordSeparators": "`~!@#$%^&*()-=+[{]}\\|;:'\",.<>/?_"` to my settings (note the final underscore) to add underscore to the default list of word separators.

This works inside the code editor, but not in places such as the search box:

![image](https://github.com/VSCodeVim/Vim/assets/3464445/b8ad19db-636a-4f3d-b91e-37951769ab7f)

Looks like it may be a bug with this setting
